File: algorithms/sorting/bubble_sort.py
# Bubble sort implementation in Python

def bubble_sort(arr: list):
    """
    Bubble sort algorithm implementation in Python
    :param arr: List of integers
    :return: List of integers
    """
    n = len(arr) # 5
    for i in range(n):
        for j in range(0, n-i-1):
            if arr[j] > arr[j+1]:
                arr[j], arr[j+1] = arr[j+1], arr[j]
    return arr

# Example usage
arr = [12, 22, 11, 25, 64]
print(bubble_sort(arr))

File: algorithms/sorting/bubble_sort.cpp
// Bubble sort in C++:

#include <iostream>
#include <vector>

using namespace std;

void bubble_sort(vector<int> &arr){
    for (int i = 0; i < arr.size(); i++){
        for (int j = 0; j < arr.size()-i-1; j++){
            if (arr[j] > arr[j+1]){
                swap(arr[j], arr[j+1]);
            }
        }
    }
}

int main(){
    vector<int> arr = {5, 2, 4, 6, 1, 3};

    bubble_sort(arr);

    for (int i = 0; i < arr.size(); i++){
        cout << arr[i] << " ";
    }

    cout << endl;

    return 0;
}